nav2_costmap_2d/plugins/costmap_filters/binary_filter.cpp

```diff
@@ -259,6 +291,65 @@ void BinaryFilter::changeState(const bool state)
     std::make_unique<std_msgs::msg::Bool>();
   msg->data = state;
   binary_state_pub_->publish(std::move(msg));
+  changeParameters(state);
+}
+
+void BinaryFilter::changeParameters(const bool state)
+{
+  for (size_t param_index = 0; param_index < binary_parameters_info_.size(); ++param_index) {
+    if (!change_parameters_clients_.at(param_index)->wait_for_service(
+        std::chrono::milliseconds(change_parameter_timeout_)))
+    {
+      RCLCPP_WARN(
+        logger_, "BinaryFilter:  service %s not available. Skipping ...",
+        change_parameters_clients_.at(param_index)->get_service_name());
+      continue;
+    } else {
+      RCLCPP_INFO(
+        logger_, "BinaryFilter:  service %s available.",
+        change_parameters_clients_.at(param_index)->get_service_name());
```

Comment:
Look ups aren't free. Store a reference to `change_parameters_clients_.at(param_index)` and use it once

Reply:
https://github.com/ros-planning/navigation2/pull/3796/commits/66b006b6ad0fd80eada2ab529a6741917f9d1b51